Source/WebCore/page/csp/ContentSecurityPolicy.cpp

```diff
@@ -810,8 +810,18 @@ bool ContentSecurityPolicy::allowMissingTrustedTypesForSinkGroup(const String& s
 
             String consoleMessage = makeString(policy->isReportOnly() ? "[Report Only] "_s : ""_s,
                 "This requires a "_s, stringContext, " value else it violates the following Content Security Policy directive: \"require-trusted-types-for 'script'\""_s);
-
-            String sample = makeString(sink, '|', source.left(40));
+            StringView adjustedSource = source;
+            if (sink == "Function"_s) {
+                if (source.startsWith("function anonymous"_s))
+                    adjustedSource = source.substring(18, source.length());
+                else if (source.startsWith("async function anonymous"_s))
+                    adjustedSource = source.substring(24, source.length());
+                else if (source.startsWith("function* anonymous"_s))
+                    adjustedSource = source.substring(19, source.length());
+                else if (source.startsWith("async function* anonymous"_s))
+                    adjustedSource = source.substring(25, source.length());
```

Comment:
I think we should do the same as `readFeatureIdentifier` in `PermissionsPolicy.cpp` does, which is roughly what you are suggesting, minus the array and for loop. Just give each string its own `constexpr` and use that for both the value and length determination.

Reply:
I think the version with the array and the loop is fantastic; I would have specified `constexpr` but otherwise it’s what I would prefer over wordier code. Unless there is some performance problem, which seems unlikely.

Reply:
Okay sounds good, I've switched the code to Tim's example but with a contexpr array.